nbhttp/client.go

```diff
@@ -133,11 +134,17 @@ func (c *httpConn) onResponse(res *http.Response, err error) {
 							c.CloseWithError(ErrClientTimeout)
 						})
 					} else {
-						c.conn.SetReadDeadline(deadline)
+						err := c.conn.SetReadDeadline(deadline)
+						if err != nil {
+							logging.Error("failed to set read dead line: %v", err)
+						}
 					}
 				}
 			} else {
-				c.conn.SetReadDeadline(time.Time{})
+				err = c.conn.SetReadDeadline(time.Time{})
+				if err != nil {
+					logging.Error("failed to set read dead line: %v", err)
+				}
```

Comment:
It will not return an error, sometimes, the lint is bad:
```golang
// SetReadDeadline implements SetReadDeadline
func (c *Conn) SetReadDeadline(t time.Time) error {
	c.mux.Lock()
	if !c.closed {
		if !t.IsZero() {
			now := time.Now()
			if c.rTimer == nil {
				c.rTimer = c.g.afterFunc(t.Sub(now), func() { c.closeWithError(errReadTimeout) })
			} else {
				c.rTimer.Reset(t.Sub(now))
			}
		} else if c.rTimer != nil {
			c.rTimer.Stop()
			c.rTimer = nil
		}
	}
	c.mux.Unlock()
	return nil
}
```